src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt

```diff
@@ -288,28 +288,41 @@ ARGS=(
 
 
 function create_and_run_classpath_jar() {
-  # Build class path as one single string separated by spaces
-  MANIFEST_CLASSPATH=""
+  # Build class path.
+  MANIFEST_CLASSPATH=()
   if is_windows; then
     CLASSPATH_SEPARATOR=";"
-    URI_PREFIX="file:/"  # e.g. "file:/C:/temp/foo.jar"
   else
     CLASSPATH_SEPARATOR=":"
-    URI_PREFIX="file:$(pwd)/"  # e.g. "file:/usr/local/foo.jar"
   fi
 
-  URI_PREFIX=${URI_PREFIX//\//\\/}
-  MANIFEST_CLASSPATH="${CLASSPATH_SEPARATOR}${CLASSPATH}"
+  OLDIFS="$IFS"
+  IFS="${CLASSPATH_SEPARATOR}" # Use a custom separator for the loop.
+  for path in ${CLASSPATH}; do
+    # Convert spaces to %20
 
-  MANIFEST_CLASSPATH=$(sed "s/ /%20/g" <<< "${MANIFEST_CLASSPATH}")
-  MANIFEST_CLASSPATH=$(sed "s/$CLASSPATH_SEPARATOR/ $URI_PREFIX/g" <<< "${MANIFEST_CLASSPATH}")
+    if is_windows; then
+      path="file:/${path}" # e.g. "file:/C:/temp/foo.jar"
```

Comment:
isn't this supposed to be `file://` (note the double slash?)

Reply:
No, the syntax is, apparently, file:(path), where path can be relative to the current directory or absolute. This matches the previous implementation.